pkg/cloudprovider/aws/initialization.go

```diff
@@ -0,0 +1,242 @@
+					"ec2.amazonaws.com"
+				  ]
+				}
+			  }
+			]
+		  }`),
+	})
+	log.PanicIfError(err, "Failed to create role %s", name)
+	zap.S().Infof("Successfully created role %s", name)
+
+	// 2. Attach policies to role
+	for _, policyArn := range []string{
+		"arn:aws:iam::aws:policy/AmazonSSMManagedInstanceCore",
+		"arn:aws:iam::aws:policy/AmazonEKSWorkerNodePolicy",
+		"arn:aws:iam::aws:policy/AmazonEKS_CNI_Policy",
+		"arn:aws:iam::aws:policy/AmazonEC2ContainerRegistryReadOnly",
+	} {
+		_, err := IAMAPI.AttachRolePolicy(&iam.AttachRolePolicyInput{
+			PolicyArn: aws.String(policyArn),
+			RoleName:  aws.String(name),
+		})
+		log.PanicIfError(err, "Failed to attach policy %s to role %s", policyArn, name)
+		zap.S().Infof("Successfully attached policy %s to role %s", policyArn, name)
+	}
+	return createRoleOutput.Role
+}
+
+func launchTemplateOrDie(ec2API ec2iface.EC2API, cluster *eks.Cluster, name string, instanceProfileName string) *ec2.LaunchTemplate {
+	// 1. Detect existing launch template
+	describeLaunchTemplateOutput, err := ec2API.DescribeLaunchTemplates(&ec2.DescribeLaunchTemplatesInput{
+		LaunchTemplateNames: []*string{aws.String(name)},
+	})
+	if aerr, ok := err.(awserr.Error); ok && aerr.Code() != "InvalidLaunchTemplateName.NotFoundException" {
+		log.PanicIfError(aerr, "Failed to retrieve default launch template")
+	}
+	if len(describeLaunchTemplateOutput.LaunchTemplates) > 0 {
+		zap.S().Infof("Successfully detected existing launch template, %s/%s",
+			*describeLaunchTemplateOutput.LaunchTemplates[0].LaunchTemplateId,
+			*describeLaunchTemplateOutput.LaunchTemplates[0].LaunchTemplateName)
+		return describeLaunchTemplateOutput.LaunchTemplates[0]
+	}
+
+	// 2. Create Launch Template
+	createLaunchTemplateOutput, err := ec2API.CreateLaunchTemplate(&ec2.CreateLaunchTemplateInput{
+		LaunchTemplateName: aws.String(name),
+		LaunchTemplateData: &ec2.RequestLaunchTemplateData{
+			IamInstanceProfile: &ec2.LaunchTemplateIamInstanceProfileSpecificationRequest{
+				Name: aws.String(instanceProfileName),
+			},
+			TagSpecifications: []*ec2.LaunchTemplateTagSpecificationRequest{{
+				ResourceType: aws.String(ec2.ResourceTypeInstance),
+				Tags: []*ec2.Tag{{
+					Key:   aws.String(fmt.Sprintf("kubernetes.io/cluster/%s", *cluster.Name)),
+					Value: aws.String("owned"),
+				}},
+			}},
+			SecurityGroupIds: append(cluster.ResourcesVpcConfig.SecurityGroupIds, cluster.ResourcesVpcConfig.ClusterSecurityGroupId),
+			UserData: aws.String(base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf(`
+				#!/bin/bash
+				yum install -y https://s3.amazonaws.com/ec2-downloads-windows/SSMAgent/latest/linux_amd64/amazon-ssm-agent.rpm
+				/etc/eks/bootstrap.sh %s \
+					--kubelet-extra-args '--node-labels=karpenter.sh/provisioned=true' \
+					--b64-cluster-ca %s \
+					--apiserver-endpoint %s`,
+				*cluster.Name,
+				*cluster.CertificateAuthority.Data,
+				*cluster.Endpoint,
+			)))),
+			ImageId: aws.String("ami-0532808ed453f9ca3"),
```

Comment:
This will not work in every region, can we add a TODO here.

Reply:
Done